pricelist_per_product/pricelist.py

```diff
@@ -91,12 +93,38 @@ def button_product(self):
         }
 
     @api.model
+    def update_sequence_price_grid(self, vals):
```

Comment:
Hmmm I'm not so sure how I feel about this method mutating the `vals`. Can we instead return a dict `{related_sequence: #}`, then use the return of this method to update the vals dict? eg
```
vals.update(
    self.update_sequence_price_grid(vals)
)
```

I know it seems backwards sort of, but I had to read the whole code block a few times in order to identify what was happening, which is a sign of too much complexity. This method doesn't actually own `vals`, so changing it isn't necessarily the best course of action.